### DISCUSSION TITLE
Exception from WebAPI calls if pocketbase is down

### DISCUSSION THREAD

**Discussion by rameshkumar70** [pocketbase/pocketbase#5563] [Q&A]
Hi,
How would I know whether the pocketbase server is down when invoking WebAPI calls such as  pb.collection('users').create(record)? Do these calls throw an exception when pocketbase server is down? I couldn't detect any exceptions to that effect.  All I get is something like this:
{"url":"","status":0,"response":{},"isAbort":false,"originalError":{},"name":"ClientResponseError 0"}

Any input is greatly appreciated. Thanks.

**Comment by calumk** [comment]
You can use /api/health 

or

Checks the health status of the api.
`pb.health.check(options = {});`


I guess you could hook into aftersend and if the response status is 0 do an api check?

```
pb.afterSend = function (response, data) {
    // do something with the response state
    console.log(response.status);

    return Object.assign(data, {
        // extend the data...
        "additionalField": 123,
    });
};
```
